internal/cloud_connector/message_handlers.go

```diff
@@ -247,8 +246,8 @@ func processDispatchers(sourcesRecorder controller.SourcesRecorder, identity dom
 	}
 }
 
-func handleOfflineMessage(client MQTT.Client, clientID domain.ClientID, msg protocol.ControlMessage, connectionRegistrar connection_repository.ConnectionRegistrar) error {
-	logger := logger.Log.WithFields(logrus.Fields{"client_id": clientID})
+func handleOfflineMessage(logger *logrus.Entry, client MQTT.Client, clientID domain.ClientID, msg protocol.ControlMessage, connectionRegistrar connection_repository.ConnectionRegistrar) error {
+	logger = logger.WithFields(logrus.Fields{"client_id": clientID})
```

Comment:
This line can be removed now as we are setting the client_id in the logger before this function is called.